source/slang/slang-diagnostic-defs.h

```diff
@@ -524,6 +524,12 @@ DIAGNOSTIC(
     Error,
     missingLayoutBindingModifier,
     "Expecting 'binding' modifier in the layout qualifier here")
+DIAGNOSTIC(
+    20017,
+    Error,
+    disallowConstAsModifierOfCppPtr,
+    "cannot use 'const' on 'T*', prefer to use the syntax 'const Ptr<T>' instead")
```

Comment:
‘const’ not allowed on pointer typed declarations using the C style ‘*’ operator. If the intent is to restrict the pointed-to value to read-only, use ‘Ptr<T, Access.Read>’ instead, if the intent is to make the pointer itself immutable, use ‘const Ptr<T>’.